VideoLocker/src/main/java/org/edx/mobile/player/Player.java

```diff
@@ -311,18 +314,14 @@ public void setPreview(Preview preview) {
             if (preview == null) {
                 return;
             }
-            preview.getHolder().addCallback(new SurfaceHolder.Callback() {
+            preview.setSurfaceTextureListener(new TextureView.SurfaceTextureListener() {
 
                 @Override
-                public void surfaceDestroyed(SurfaceHolder holder) {
-                    // nothing to be done here
-                }
-
-                @Override
-                public void surfaceCreated(SurfaceHolder holder) {
+                public void onSurfaceTextureAvailable(SurfaceTexture surfaceTexture, int width, int height) {
                     try {
                         logger.debug("Player state=" + state);
-                        setDisplay(holder);
+                        Surface surface = new Surface(surfaceTexture);
+                        setSurface(surface);
                         // keep screen ON
                         setScreenOnWhilePlaying(true);
```

Comment:
Method <code>setScreenOnWhilePlaying(boolean)</code> works only when using <code>SurfaceView</code> and <code>SurfaceHolder</code>. So, when using <code>TextureView</code> this method actually does nothing.

The implementation for keeping the app awake is completed on #179 .